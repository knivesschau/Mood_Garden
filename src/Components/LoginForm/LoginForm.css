/* mobile first style, min 320px */

.Login_Form {
    display: flex;
    flex-direction: column;
    font-weight: bold;
    text-align: center;
    margin: 1em;
    padding: 1em;
    font-size: 1em;
    background-color: rgb(255, 152, 169, .80);
    box-shadow: inset -4px 1px 20px 2px #A8739B;
    border: 1px transparent white;
    border-radius: 30em;
}

.Login_Nav {
    font-size: 1.25em;
    text-transform: uppercase;
    font-weight: bold;
    display: inline-block;
    padding: 0.2em .5em;
    margin: 0 0.1em 0.1em 0;
    border: 2px solid #FFFFFF;
    border-radius: 1em;
    box-sizing: border-box;
    text-shadow: 0 .65px 0.65px rgb(0,0,0,0.35);
    transition-property: background;
    transition-duration: .5s;
    transition-timing-function: ease-in;
}

.Login_Nav:hover {
    cursor: pointer;
    background-color: #FFADBB;
    color: #FFFFFF;
}

.Login_Nav:active {
    background-color: #B00000;
    color: #FFFFFF;
}

#login-header {
    text-align: center;
    font-size: 1.5em;
    margin-top: 1em;
    margin-bottom: .5em;
    text-shadow: 1px 0px 0px #000000;
}

#return-user-label, #return-pass-label {
    margin-top: 1em;
    margin-bottom: 1em;
}

#submit-login {
    display: inline-block;
    padding: 0.3em 1.2em;
    margin: 4em 4em 0.1em 4em;
    border: 2px solid rgba(255,255,255,0);
    border-radius: 2em;
    box-sizing: border-box;
    margin-top: 1em;
    background-color: #C34444;
    color: #ffffff;
    font-weight: bold;
    font-size: 1em;
    text-shadow: 0 .65px 0.65px rgb(0,0,0,0.35);
}

#submit-login:hover {
    cursor: pointer;
    border-color: rgb(255,255,255, 1);
}

#submit-login:active {
    background-color: #8F0000;
}

@media only screen and (min-device-width: 375px) {
    .Login_Form {
        margin: 2em;
    }
}

@media only screen and (min-device-width: 414px) {
    .Login_Nav {
        font-size: 1.50em;
    }
    
    #login-header {
        font-size: 1.75em;
    }

    #user-descrip, #confirm-descrip, #password-descrip, #return-user-label, #return-pass-label, #submit-login {
        font-size: 1.10em;
    }
    
}

