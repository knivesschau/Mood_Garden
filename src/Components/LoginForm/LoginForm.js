import React, { Component } from "react";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import TokenService from '../../services/token-services';
import AuthApiService from '../../services/auth-api-service';
import { Link } from 'react-router-dom';
import './LoginForm.css';

export default class LoginForm extends Component { 
    static defaultProps = {
        onValidLogin: () => {}
    };

    constructor(props) {
        super(props);
        this.state = {error: null};
    };

    // handle login authentication and validation on submit. //
    handleJwtLoginAuth = e => {
        e.preventDefault();
        const {return_user, return_pass} = e.target;

        this.setState({
            error: null
        });

        AuthApiService.postLogin({
            user_name: return_user.value,
            password: return_pass.value 
        })
            .then(res => {
                return_user.value = '';
                return_pass.value = '';
                TokenService.saveAuthToken(res.authToken);
                this.props.onValidLogin();
            })
            .then(() => {
                window.location=`/your-garden`;
            })
            .catch(res => {
                this.setState({
                    error: alert("Invalid username or password. Please double-check your credentials.")
                });
            });
    }

    render() {
        return (
            <section className="GardenLoginForm">

                <nav role="navigation" className="Login_Nav"> 
                    <FontAwesomeIcon icon="chevron-left"/>
                        <Link to='/'>
                        Back 
                        </Link>
                </nav>
            
                <h3 id="login-header">Log In</h3>

                <form className="Login_Form" onSubmit={this.handleJwtLoginAuth}> 

                    <div>

                        <label htmlFor="username">
                            <p id="return-user-label">Username:</p>

                        <input type="text" id="return-user" name="return_user"/>

                        </label>

                    </div>

                    <div>

                        <label htmlFor="username">
                            <p id="return-pass-label">Password:</p>
                            
                        <input type="password" id="return-pass" name="return_pass"/>

                        </label>
                    </div>

                    <button type="submit" id="submit-login">Login</button>
                
                </form>

                <div className="Demo_Info">

                    <span id="demo-info1">Demo User: thinkfulTeam2020</span>

                    <span id="demo-info2">Demo Password: M0odGarden1!</span>

                </div>

            </section>
        );
    }
}