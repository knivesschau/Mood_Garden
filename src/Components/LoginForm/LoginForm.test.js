import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginForm from './LoginForm';

describe ('LoginForm Component', () => {
    it ('renders the complete LoginForm Component', () => {
        const wrapper = shallow(<LoginForm/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})