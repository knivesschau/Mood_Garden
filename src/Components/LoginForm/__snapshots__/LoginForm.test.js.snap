// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component renders the complete LoginForm Component 1`] = `
<section
  className="LoginForm"
>
  <nav
    className="Login_Nav"
    role="navigation"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="chevron-left"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <Link
      to="/"
    >
      Back
    </Link>
  </nav>
  <h3
    id="login-header"
  >
    Log In
  </h3>
  <form
    className="Login_Form"
    onSubmit={[Function]}
  >
    <div>
      <label
        htmlFor="username"
      >
        <p
          id="return-user-label"
        >
          Username:
        </p>
        <input
          id="return-user"
          name="return_user"
          type="text"
        />
      </label>
    </div>
    <div>
      <label
        htmlFor="username"
      >
        <p
          id="return-pass-label"
        >
          Password:
        </p>
        <input
          id="return-pass"
          name="return_pass"
          type="password"
        />
      </label>
    </div>
    <button
      id="submit-login"
      type="submit"
    >
      Login
    </button>
  </form>
</section>
`;
